data/plugins/language_lua.lua

```diff
@@ -27,9 +27,53 @@ syntax.add {
     { pattern = "%.%.%.?",                   type = "operator" },
     { pattern = "[<>~=]=",                   type = "operator" },
     { pattern = "[%+%-=/%*%^%%#<>]",         type = "operator" },
-    { pattern = "[%a_][%w_]*()%s*%f[(\"'{]", type = {"function", "normal"} },
-    { pattern = "[%a_][%w_]*",               type = "symbol" },
+    -- Metamethods
+    { pattern = "__[%a][%w_]+",              type = "function" },
+    { pattern = "%.()__[%a][%w_]+",
+      type = { "normal", "function" }
+    },
+    -- Variable declaration
+    { pattern = "local()%s+()[%a_][%w_]+",
+      type = { "operator", "normal", "keyword2" }
+    },
+    -- Multiple variable declarations
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*() *=",
+      type = { "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*=",
+      type = { "keyword2", "normal", "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*=",
+      type = { "keyword2", "normal", "keyword2", "normal", "keyword2", "normal", "keyword2", "operator" }
+    },
+    { pattern = "[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*()%s*,%s*()[%a][%w_]*%s*=",
```

Comment:
Could this be done in a saner way?
Maybe something like
```lua
    { pattern = {"local", "[=\n]"},
      syntax = {
        patterns = {
          {
            pattern = "[%a_][%w_]*()%s*,?%s*",
            type = {"keyword2", "normal"}
          }
        },
        symbols = {}
      },
      type = "normal"
    },
```

Reply:
ahh, interesting solution! will test to see if it works

Reply:
this causes the `=` to not be highlighted as a keyword anymore, change the type to operator?